lms/djangoapps/teams/tests/test_views.py

```diff
@@ -1666,6 +1913,40 @@ def test_student_join_instructor_managed_team(self, *args):  # pylint: disable=u
             self.build_membership_data_raw(self.users['student_enrolled_not_on_team'].username, self.solar_team.team_id)
         )
 
+    @ddt.data(
+        ('student_masters', 400, 'is already a member'),
+        ('student_masters_not_on_team', 200, None),
+        ('student_unenrolled', 404, None),
+        ('student_enrolled', 403, None),
+        ('student_enrolled_both_courses_other_team', 403, None),
+        ('staff', 200, None),
+    )
+    @ddt.unpack
+    def test_join_organization_protected_team(self, user, expected_status, expected_message):
+        response = self.post_create_membership(
```

Comment:
student_enrolled' and 'student_enrolled_both_courses_other_team' should not be able to see that the team exists at all